### DISCUSSION TITLE
IPv6 only servers

### DISCUSSION THREAD

**Discussion by komandar** [vitobotta/hetzner-k3s#213] [Ideas]
The current version always creates servers with IPv4 and IPv6 IPs.
Hetzner offers servers without IPv4 and you save the costs for this.

For our current projects we only need one IPv6 IP which is why it would be great if you could configure in the script whether you need an IPv4 or not.

**Comment by derlinuxer** [comment]
Yes would be great to disable IPv4 & IPv6 external Interface at node creation.

**Comment by vitobotta** [comment]
Added to my list.

**Comment by derlinuxer** [comment]
@vitobotta I did some changes to your code. So it's now possible to disable public interfaces. ipv4 or ipv6 or both. I'll write some words to the README and will send you a PR. I'm not familiar with crystal or your programming concept. So there are maybe some side effects based on the changes. I guess you see that very fast. It works absolut find in my case (disabled both and set proper net Config via cloud init). Fully cluster ist running in hetzner private network, of course the load balancer which has external IP.

**Reply by vitobotta** [reply]
Sounds awesome, thanks! Please send the PR. I am quite busy at the moment so it may take a little time before I have a chance to review and test/merge, but will do it ASAP.
